### DISCUSSION TITLE
Map View gray on Android

### DISCUSSION THREAD

**Discussion by minifisk** [react-native-maps/react-native-maps#4773] [Q&A]
### Summary

I've looked at previous threads and followed this tutorial:
https://docs.expo.dev/versions/latest/sdk/map-view/#deploying-google-maps-to-an-android-standalone

But still the map is gray on Android.

This is the relevant details:

The code in my expo app:
```
import MapView, { Marker, Callout, PROVIDER_GOOGLE, PROVIDER_DEFAULT } from "react-native-maps";

<MapView
        provider={Platform.OS === 'android' ? PROVIDER_GOOGLE : PROVIDER_DEFAULT}
...
</MapView>
```

The code in my app.json
   ```
 "android": {
      "versionCode": 8,
      "config": {
        "googleMaps": {
          "apiKey": "xxxxxx"
        }
      },
```

I then create an APK development build, download it to my Android phone, start the server with expo start -c, open the app, and the map view is gray with the Google logo in the bottom.

I've tried creating a new API-key, rebuilding the APK build, and setting the API-key in its Android restrictions with the bundle ID and the SHA-1 key as Fingerprint, but still having the same issue.

I also have Maps SDK for Android under Enabled API's & Services in the cloud console.

What am I missing?

### Reproducible sample code

```TSX
<MapView
        provider={Platform.OS === 'android' ? PROVIDER_GOOGLE : PROVIDER_DEFAULT}
...
</MapView>
```

### Steps to reproduce

Check summary

### Expected result

Check summary

### Actual result

Check summary

### React Native Maps Version

1.7.1

### What platforms are you seeing the problem on?

Android

### React Native Version

0.71.8

### What version of Expo are you using?

SDK 48

### Device(s)

Samsung Galaxy S6 Edge

**Comment by minifisk** [comment]
No one?

**Reply by ssommelet21** [reply]
Hi minifisk.
Do you solve your problem ? because i don't have problem using "expo" & "react-native-maps" & "eas build -p android --profile preview" (to generate an apk).

Maybe you miss something, here is my android section

    ...
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "permissions": [
        "android.permission.READ_EXTERNAL_STORAGE",
        "android.permission.WRITE_EXTERNAL_STORAGE",
        "android.permission.ACCESS_MEDIA_LOCATION",
        "android.permission.ACCESS_COARSE_LOCATION",
        "android.permission.ACCESS_FINE_LOCATION",
        "android.permission.FOREGROUND_SERVICE",
        "android.permission.ACCESS_BACKGROUND_LOCATION",
        "android.permission.INTERNET"
      ],
      "config": {
        "googleMaps": {
          "apiKey": "AIzaSyxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        }
      },
      "package": "com.abc.def"
    },
    
  Do you see in google api key section, Maps SDK for Android counter increase ? run after run ?

**Comment by samuelpismel** [comment]
I had the same problem here. 

Even after following the API configurations of this link https://docs.expo.dev/versions/latest/sdk/map-view/#deploy-app-with-google-maps the problem persisted.

It only happened in the **preview build** (made with `eas build --profile preview --platform android --clear-cache `).

My solution was create a production build (.aab extension) and upload as **Internal Test** on **Google Play Console**. 
(Remember to add you email as tester at Users Tab)

After some minutes I've downloaded this "beta build" via Google Play and the MapView was indeed working when released by Google Play.

The problem was the API Key restrictions (sha-1 signature). 
It can be configured here:  https://console.cloud.google.com/apis/credentials
_API Key Menu > Edit Api Key > Key restrictions > Application restrictions_
and
_API Key Menu > Edit Api Key > Key restrictions > Restrict usage to your Android apps_

I hope it works for you. :)
